lua/cmp_nvim_lsp/init.lua

```diff
@@ -20,29 +20,38 @@ local if_nil = function(val, default)
   return val
 end
 
-M.update_capabilities = function(capabilities, override)
+
+M.default_capabilities = function(override)
   override = override or {}
 
-  local completionItem = capabilities.textDocument.completion.completionItem
-
-  completionItem.snippetSupport = if_nil(override.snippetSupport, true)
-  completionItem.preselectSupport = if_nil(override.preselectSupport, true)
-  completionItem.insertReplaceSupport = if_nil(override.insertReplaceSupport, true)
-  completionItem.labelDetailsSupport = if_nil(override.labelDetailsSupport, true)
-  completionItem.deprecatedSupport = if_nil(override.deprecatedSupport, true)
-  completionItem.commitCharactersSupport = if_nil(override.commitCharactersSupport, true)
-  completionItem.tagSupport = if_nil(override.tagSupport, { valueSet = { 1 } })
-  completionItem.resolveSupport = if_nil(override.resolveSupport, {
-    properties = {
-      'documentation',
-      'detail',
-      'additionalTextEdits',
-    }
-  })
-
-  return capabilities
+  return {
+    textDocument = {
+      completion = {
+        snippetSupport = if_nil(override.snippetSupport, true),
+        preselectSupport = if_nil(override.preselectSupport, true),
+        insertReplaceSupport = if_nil(override.insertReplaceSupport, true),
+        labelDetailsSupport = if_nil(override.labelDetailsSupport, true),
+        deprecatedSupport = if_nil(override.deprecatedSupport, true),
+        commitCharactersSupport = if_nil(override.commitCharactersSupport, true),
+        tagSupport = if_nil(override.tagSupport, { valueSet = { 1 } }),
+        resolveSupport = if_nil(override.resolveSupport, {
+            properties = {
+                "documentation",
+                "detail",
+                "additionalTextEdits",
+            },
+        }),
+      },
+    },
+  }
+end
+
+---Backwards compatibility
+M.update_capabilities = function(capabilities, override)
+  return M.default_capabilities(override)
```

Comment:
I added the deprecating warning and for the version when the function will be removed filled in `1.0.0`. If you think that's good than I think this is resolved, but I don't know what version you want since this plugin doesn't have a version?

Reply:
This is a breaker! `vim.deprecate` shall not be used, as vim.deprecate is not available as of 0.8.0. Please test more thoroughly with stable versions. I've made a PR.

Reply:
I think 0.8.0 is released and stable version.

Reply:
Yes. vim.deprecate is not included in 0.8.0; it is only available in the nightly as of now.